### PR TITLE
fix(codewhisperer): Re-enable the suggestion navigation after 1.78 VS Code inline completion API breaking changes

### PR DESCRIPTION
**Pull request by leigaol**
## Problem

1. Suggestion navigation is no longer working when using VS Code 1.78+
2. Users need to hit Esc twice to clear/reject a code suggestion.

Root cause:

The dispose method of a inlineCompletionProvider used to clear the inline Completion widget, which allows us to update the widget whenever we want to by doing a dispose and doing another register with a new inlineCompletionProvider. However, after VS Code 1.78+, disposing the provider no longer clears the active inline completion widget, which is a regression because disposing a object should clear the states it created.

## Solution

Manually refresh the suggestion ghost text after each dispose of `inlineCompletionProvider` only for VS Code 1.78+.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/codewhisperer/service/inlineCompletionService.ts [hunk 1]
```diff
@@ -18,13 +18,15 @@ import { getLogger } from '../../shared/logger/logger'
...
-import { getPrefixSuffixOverlap } from '../util/commonUtil'
+import { getPrefixSuffixOverlap, isVscHavingRegressionInlineCompletionApi } from '../util/commonUtil'
 import globals from '../../shared/extensionGlobals'
 import { AuthUtil } from '../util/authUtil'
 import { shared } from '../../shared/utilities/functionUtils'
 import { ImportAdderProvider } from './importAdderProvider'
+import { Mutex } from 'async-mutex'
 
 const performance = globalThis.performance ?? require('perf_hooks').performance
+const mutex = new Mutex()
```
Comment (JadenSimon) [inline src/codewhisperer/service/inlineCompletionService.ts:29]: Use `async-lock` instead. Adding a new dependency is unnecessary.

Comment (leigaol) [reply src/codewhisperer/service/inlineCompletionService.ts:29]: ACK. I will update it in the next commit.

Comment (leigaol) [reply src/codewhisperer/service/inlineCompletionService.ts]: I have switched to async-lock instead.

**Review by JadenSimon — APPROVED** [review]
Needs changelog entry right?

**Comment by leigaol** [comment]
> Needs changelog entry right?

Yes, let me add one.

**Review by zixlin7 — APPROVED** [review]

**Review by JadenSimon — APPROVED** [review]

File: .changes/next-release/Bug Fix-65050370-fd7c-49e6-86ba-00766572400c.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "CodeWhisperer: Fix a bug when user cannot navigate to next or previous code suggestion in VS Code 1.78+"
```
Comment (justinmk3) [inline .changes/next-release/Bug Fix-65050370-fd7c-49e6-86ba-00766572400c.json:3]:
```suggestion
	"description": "CodeWhisperer: cannot navigate to next or previous code suggestion in VS Code 1.78+"
```

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**